src/compiler/propagation/type_propagator.cc

```diff
@@ -254,7 +254,7 @@ void TypePropagator::propagate(TypeDatabase* types) {
   sites_.for_each([&](uint8* site, Set<TypeVariable*>& results) {
     type.clear(words_per_type());
     for (auto it = results.begin(); it != results.end(); it++) {
-      type.add_all((*it)->type(), words_per_type());
+      type.add_all_even_blocks((*it)->type(), words_per_type());
```

Comment:
It took me way too long to realize this is not talking about "even blocks", where "even" means not odd.
```suggestion
      type.add_all_also_blocks((*it)->type(), words_per_type());
```

Reply:
Done.